lib/add_magic_comment.rb

```diff
@@ -16,11 +16,20 @@ module AddMagicComment
 
   def self.process(argv)
     directory = argv.first || Dir.pwd
+    ignore_pattern = Regexp.new(argv[1]) rescue nil
+    # Allow providing as either a Regex or a String
+    if ignore_pattern && ignore_pattern.is_a?(String)
```

Comment:
Ohhhhh, OK I see what you're getting at. Yes, that is a bad refactoring. It should be
   
    ignore_pattern = argv[1] rescue nil

I'll fix it up and re-push.